File: 2099-find-subsequence-of-length-k-with-the-largest-sum/2099-find-subsequence-of-length-k-with-the-largest-sum.cpp
class Solution {
public:
    vector<int> maxSubsequence(vector<int>& nums, int k) {
        
        vector<int>nums2=nums,ans;
        sort(nums2.begin(),nums2.end());
       for(int i=nums2.size()-k;i<nums2.size();i++) ans.push_back(nums2[i]);
        vector<int>res;
        for(int i=0;i<nums.size();i++){
             auto it = find(ans.begin(), ans.end(),nums[i]);
             if (it != ans.end()) {
              res.push_back(nums[i]);
              ans.erase(it);
             }
        }
    
        return res;
    }
};

File: 2099-find-subsequence-of-length-k-with-the-largest-sum/2099-find-subsequence-of-length-k-with-the-largest-sum.java
class Solution {
    public int[] maxSubsequence(int[] nums, int k) {
        int [] ans=new int[k];
        int n=nums.length;
        PriorityQueue<int[]>pq=new PriorityQueue<>((a,b)->a[0]-b[0]);
        for(int i=0;i<n;i++){
            pq.add(new int[]{nums[i],i});
            if(pq.size()>k) pq.poll();
        }
        Set<Integer> st=new HashSet<>();
        while (pq.size()>0){
            st.add(pq.poll()[1]);
        }
        int index=0;
        for(int i=0;i<n;i++){
            if(st.contains(i)) ans[index++]=nums[i];
        }
        return ans;
    }
}